### PR TITLE
pdpb: add replicate_mode

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

See PD issue https://github.com/pingcap/pd/issues/2272 for detail.

### CODE REVIEW

File: proto/replicate_mode.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package replicate_mode;
+
+// The replicate status sync from PD to TiKV.
+message ReplicateStatus {
+    enum Mode {
+        MAJORITY = 0;  // the standard mode. Replicate logs to majority peer.
+        MS_AUTOSYNC = 1; // master-slave mode. Replicate logs among 2 DCs.
```
Comment (BusyJay) [inline proto/replicate_mode.proto]: How about "PS_AUTO"? "PS" stands for primary and secondary. Or "Active-Active".

File: proto/replicate_mode.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// The status of ms-autosync mode.
+message MSAutoSync {
+    enum State {
+        SYNC = 0; // raft logs need to sync between different DCs
+        ASYNC = 1; // raft logs need to sync to majority peers
+        SYNC_RECOVER = 2; // switching from ASYNC to SYNC mode
+    }
+    string label = 1; // the key of the label that used for distinguish different DC.
```
Comment (BusyJay) [inline proto/replicate_mode.proto]: How about "label_key"?

File: proto/replicate_mode.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    State state = 2;
+    uint64 recover_id = 3; // Unique ID for a recovery
+    int32 wait_sync_timeout_hint = 4; // Duration to wait before switching to SYNC by force
```
Comment (BusyJay) [inline proto/replicate_mode.proto]: Should mention its unit.

File: proto/replicate_mode.proto (hunk 1 above)
Comment (disksing) [reply proto/replicate_mode.proto]: seems PS is harder for reader to guess its meaning. I'm thinking maybe just AUTOSYNC

File: proto/replicate_mode.proto (hunk 2 above)
Comment (disksing) [reply proto/replicate_mode.proto]: good idea

File: proto/replicate_mode.proto (hunk 3 above)
Comment (disksing) [reply proto/replicate_mode.proto]: good idea

File: proto/replicate_mode.proto [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+// The replicate status sync from TiKV to PD.
+message RegionReplicateStatus {
+    enum State {
+        MAJORITY = 0; // Logs sync to majority peers
+        INTEGRITY_OVER_LABEL = 1; // Logs sync to different DCs
+    }
+    State state = 1;
+    uint64 recover_id = 2; // Unique ID for a recovery
+}
```
Comment (HunDunDM) [inline proto/replicate_mode.proto]: Need new blank line？

Comment (disksing) [reply proto/replicate_mode.proto]: good idea

**Comment by disksing** [comment]
PTAL @BusyJay @nolouch @HunDunDM

File: proto/replicate_mode.proto [hunk 5]
```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package replicate_mode;
+
+// The replicate status sync from PD to TiKV.
+message ReplicateStatus {
```
Comment (BusyJay) [inline proto/replicate_mode.proto:5]: Maybe we need to name it as `ReplicationStatus`?

Comment (disksing) [reply proto/replicate_mode.proto:5]: I think it ok. You can send a PR.

**Pull request merged**